homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -68,18 +67,25 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the FRITZ!SmartHome binary sensor from ConfigEntry."""
-    coordinator: FritzboxDataUpdateCoordinator = hass.data[FRITZBOX_DOMAIN][
-        entry.entry_id
-    ][CONF_COORDINATOR]
-
-    async_add_entities(
-        [
-            FritzboxBinarySensor(coordinator, ain, description)
-            for ain, device in coordinator.data.devices.items()
-            for description in BINARY_SENSOR_TYPES
-            if description.suitable(device)
-        ]
-    )
+    coordinator = get_coordinator(hass, entry.entry_id)
+
+    @callback
+    def _add_entities() -> None:
+        """Add devices."""
+        if not coordinator.new_devices:
+            return
+        async_add_entities(
+            [
```

Comment:
We can remove the brackets to make it a generator expression instead of a list comprehension.

Reply:
#104267